engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnAlertsServiceImpl.java

```diff
@@ -715,12 +721,12 @@ public Event getEvent(String tenantId, String eventId, boolean thin) throws Exce
             }
         }
 
-        List<IspnEvent> ispnEvents = getEventItems(query, pager);
+        Page<IspnEvent> ispnEvents = getEventItems(query, pager);
         List<Event> events = ispnEvents.stream().map(e -> e.getEvent()).collect(Collectors.toList());
         if (events.isEmpty()) {
-            return new Page<>(events, pager, 0);
+            return new Page<>(events, pager, ispnEvents.getTotalSize());
```

Comment:
Same question as for the alerts - is there a real difference in these paths? Makes reading code more difficult if one jumps to a function and one doesn't.

Reply:
Also, should we have a test where the return is empty? Two paths or not - it seems to indicate there's a difference in handling them, but that other case isn't tested. Will it break something?

Reply:
This is tested when for the 1000 items case it looks at page 20 (of size 50), which is beyond the end.